content/blog/migrating-vsphere-virtual-volumes-to-kubevirt-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,432 @@
+| mariadb-kubevirt |
++------------------+:
+ok: [mariadb-kubevirt]
+
+PLAY RECAP *************************************************************************
+mariadb-kubevirt           : ok=6    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+
+Success! The cloning process worked. Now, before finalizing with the actual volume. Let’s run a “detach” playbook to stop the database and unmount the filesystem.
+
+```yaml
+---
+# With PVC attached to the KubeVirt VM, stop MariaDB and unmount the filesystem
+- hosts: kubevirt
+  become: yes
+  tasks:
+  - name: Make sure MariaDB is stopped and disabled
+    service:
+      name: mariadb
+      state: stopped
+      enabled: no
+
+  - name: Remove data disk mounts
+    mount:
+      path: "{{ var_lib_mysql }}"
+      state: absent
+```
+
+In the OKD UI, detach the volume and delete the PVC.
+
+![](/img/detach-disk.png "Detach disk from VM")
+
+```shell
+$ kubectl delete -f pvc-clone.yaml
+```
+
+Next, let’s migrate the workload permanently.
+
+# Finalizing
+
+The last step involves simply shutting down the source vSphere VM to ensure the source volume isn’t accessed. Once the VM is properly shutdown. Create a PVC with the “importVolumeName” annotation to move the volume out of the vVol datastore and make it into a PV.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: testdb-migrated
+  namespace: hpe-vmi
+  annotations:
+    csi.hpe.com/importVolumeName: dat-mariadbl-56480669
+spec:
+  volumeMode: Block
+  storageClassName: legacy-migration
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 128Gi
+```
+
+Create the PVC.
+
+```shell
+$ kubectl create -f pvc-migrate.yaml
+```
+
+Now, repeat the step within OKD to attach the PVC as a disk to the running VM. Once attached, run the “migrate” playbook.
+
+```shell
+...
++---------+--------+
+| summary | result |
++---------+--------+
+| CRC     | OK     |
+| count   | OK     |
++---------+--------+
++------------------+
+| VM               |
++------------------+
+| mariadb-kubevirt |
++------------------+:
+ok: [mariadb-kubevirt]
+
+PLAY RECAP *************************************************************************
+mariadb-kubevirt           : ok=6    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+
+We should see identical results as during the cloning procedure.
+
+Next, delete the source VM in vCenter and plan the next migration.
+
+# Learn more
+
+We’ve barely scratched the surface of the HPE CSI Driver for Kubernetes features with HPE Alletra Storage MP B10000. The “allowOverrides” construct along with “importVolAsClone” and “importVolumeName” lends itself to many different use cases where data migration is one of them. Importing a “foreign” volume to Kubernetes enables sophisticated DevOps processes where terabytes of data can be made available to developers, data analysts and testing frameworks using CI/CD pipelines.
```

Comment:
Change "We've" to "I've".

I'd also recommend placing an Oxford comma after "data analysts".